app/controllers/follows_controller.rb

```diff
@@ -68,7 +68,11 @@ def follow_params
   end
 
   def follow(followable, need_notification: false)
-    user_follow = current_user.follow(followable)
+    begin
+      user_follow = current_user.follow(followable)
+    rescue ActiveRecord::RecordInvalid => e
+      Rails.logger.error(e)
+    end
     Notification.send_new_follower_notification(user_follow) if need_notification
```

Comment:
I think we should add this to our happy path/begin block since we know if the follow is invalid we dont need to send a notification.